Python/pywarpx/picmi.py

```diff
@@ -1938,6 +1938,17 @@ class FieldDiagnostic(picmistandard.PICMI_FieldDiagnostic, WarpXDiagnosticBase):
 
     warpx_dump_rz_modes: bool, optional
         Flag whether to dump the data for all RZ modes
+
+    warpx_particle_fields_to_plot: dict, optional
```

Comment:
I don't know if this would be better, but perhaps instead of a dictionary, accept a list of instances of a class with the predefined attributes. Something like this
```
from dataclasses import dataclass

@dataclass(frozen=True)
class ParticleFieldDiagnostic:
    name: str
    func: str
    do_average: int = 1
    filter: str = None
```
This will let Python do some of the error checking, making sure that the input is correct. The "frozen=True" prevents extra attributes from being added.

Reply:
Hi David, 

Sorry for the late response again (I'm going to dedicate today and tomorrow to close this out). I think thats a great idea!

The majority of the classes in `picmi.py` are derived from the PICMI standard or from the `WarpXDiagnosticBase` class, and I don't think that this fits into either of these categories. Would it be fine for this to be a standalone class with just an `__init__ `function?

Reply:
Using a stand alone class for this would be good. Note that a doc string should be added (I left it out of my example for simplification).

A future PR might add this to the picmi standard, but get it working in WarpX first.

Reply:
Thank you for the suggestion, its working with the class implementation.